src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,92 +48,1243 @@
     public String groupByTeam = null;
     public String showStats = null;
     public int startIndex;
+    private boolean shouldCollapsed;
+    
+    private FieldValidator validator = new FieldValidator();
+
 
     // used for html table ajax loading
     public String courseId = null;
     public String feedbackSessionName = null;
     public String ajaxStatus = null;
     public String sessionResultsHtmlTableAsString = null;
+    
 
+    // TODO multiple page data classes for each view type
+    
+    // for question view
+    List<InstructorResultsQuestionTable> questionPanels;
+    // for giver > question > recipient, recipient > question > giver...
+    LinkedHashMap<String, InstructorFeedbackResultsSectionPanel> sectionPanels;
+    
+    enum ViewType {
+        QUESTION, GIVER_QUESTION_RECIPIENT, RECIPIENT_QUESTION_GIVER;
+        
+        public String toString() {
+            return name().toLowerCase().replaceAll("_", "-");
+        }
+    }
+    
+    
     public InstructorFeedbackResultsPageData(AccountAttributes account) {
         super(account);
         startIndex = -1;
     }
+    
+    public void initForViewByQuestion() {
+        final ViewType viewType = ViewType.QUESTION;
+        
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionToResponseMap = bundle.getQuestionResponseMap();
+        questionPanels = new ArrayList<InstructorResultsQuestionTable>();
+        
+        for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> entry : questionToResponseMap.entrySet()) {
+            FeedbackQuestionAttributes question = entry.getKey();
+            List<FeedbackResponseAttributes> responses = entry.getValue();
+            
+            questionPanels.add(buildQuestionTableAndResponseRows(question, responses, viewType, ""));
+        }
+        
+    }
+    
+    /**
+     * Creates {@code InstructorFeedbackResultsSectionPanel}s for sectionPanels.
+     * 
+     * Iterates through the responses and creates panels and questions for them. Keeps track 
+     * of missing sections, teams and participants who do not have responses 
+     * and create panels for these missing sections, teams and participants.
+     * 
+     * TODO: simplify the logic in this method
+     */
+    public void initForViewByGiverQuestionRecipient() {
+  
+        if (!bundle.isComplete) {
+            // results page to be loaded by ajax instead 
+            buildSectionPanelsForForAjaxLoading(sections);
+            return;
+        }
+        
+        if (bundle.responses.isEmpty()) {
+            // no responses, nothing to initialize
+            return;
+        }
+        
+        setShouldCollapsed(bundle.responses.size() > 500);
```

Comment:
Extract 500 to a constant? Or maybe the entire expression as a method. (Or both.)